player/lua/stats.lua

```diff
@@ -0,0 +1,734 @@
+
+local format = string.format
+local max = math.max
+local min = math.min
+
+-- Function used to record performance data
+local recorder = nil
+-- Timer used for redrawing (toggling) and clearing the screen (oneshot)
+local display_timer = nil
+-- Current page and <page key>:<page function> mappings
+local curr_page = o.key_page_1
+local pages = {}
+-- Save these sequences locally as we'll need them a lot
+local ass_start = mp.get_property_osd("osd-ass-cc/0")
+local ass_stop = mp.get_property_osd("osd-ass-cc/1")
+-- Ring buffers for the values used to construct a graph.
+-- .pos denotes the current position, .len the buffer length
+-- .max is the max value in the corresponding buffer
+local vsratio_buf, vsjitter_buf
+local function init_buffers()
+    vsratio_buf = {0, pos = 1, len = 50, max = 0}
+    vsjitter_buf = {0, pos = 1, len = 50, max = 0}
+end
+-- Save all properties known to this version of mpv
+local property_list = {}
+for p in string.gmatch(mp.get_property("property-list"), "([^,]+)") do property_list[p] = true end
+-- Mapping of properties to their deprecated names
+local property_aliases = {
+    ["decoder-frame-drop-count"] = "drop-frame-count",
+    ["frame-drop-count"] = "vo-drop-frame-count",
+    ["container-fps"] = "fps",
+}
+
+
+-- Return deprecated name for the given property
+local function compat(p)
+    while not property_list[p] and property_aliases[p] do
+        p = property_aliases[p]
+    end
+    return p
+end
+
+
+local function set_ASS(b)
+    if not o.use_ass or o.persistent_overlay then
+        return ""
+    end
+    return b and ass_start or ass_stop
+end
+
+
+local function no_ASS(t)
+    return set_ASS(false) .. t .. set_ASS(true)
+end
+
+
+local function b(t)
+    return o.b1 .. t .. o.b0
+end
+
+
+local function it(t)
+    return o.it1 .. t .. o.it0
+end
+
+
+local function text_style()
+    if not o.use_ass then
+        return ""
+    end
+    if o.custom_header and o.custom_header ~= "" then
+        return set_ASS(true) .. o.custom_header
+    else
+        return format("%s{\\r}{\\an7}{\\fs%d}{\\fn%s}{\\bord%f}{\\3c&H%s&}" ..
+                      "{\\1c&H%s&}{\\alpha&H%s&}{\\xshad%f}{\\yshad%f}{\\4c&H%s&}",
+                      set_ASS(true), o.font_size, o.font, o.border_size,
+                      o.border_color, o.font_color, o.alpha, o.shadow_x_offset,
+                      o.shadow_y_offset, o.shadow_color)
+    end
+end
+
+
+local function has_vo_window()
+    return mp.get_property("vo-configured") == "yes"
+end
+
+
+local function has_video()
+    local r = mp.get_property("video")
+    return r and r ~= "no" and r ~= ""
+end
+
+
+local function has_audio()
+    local r = mp.get_property("audio")
+    return r and r ~= "no" and r ~= ""
+end
+
+
+local function has_ansi()
+    local is_windows = type(package) == 'table'
+        and type(package.config) == 'string'
+        and package.config:sub(1, 1) == '\\'
+    if is_windows then
+        return os.getenv("ANSICON")
```

Comment:
Even without ansicon (or conemu) mpv knows to translate some ansi escape sequences on windows too. IIRC at the very least basic colors, clear line, and move cursor to the beginning of the line.

If you need a sequence which mpv doesn't support on windows, it could be simple to add.

Reply:
This was added by @atomnuker IIRC. I have no way to test on Windows.